int32 numToCapture
bool blockUntilDone
---
